src/main/resources/templates/datewarehouse/etl-jobs-config.json

```diff
@@ -55,30 +160,107 @@
 		"sourcePath" : "log.sagebase.org/nodeSnapshots/",
 		"tableDescriptor": {
 			"name": "nodesnapshots",
-			"columns": {
-				"change_type": "string",
-				"change_timestamp": "timestamp",
-				"change_user_id": "bigint",
-				"snapshot_timestamp": "timestamp",
-				"id": "bigint",
-				"benefactor_id": "bigint",
-				"project_id": "bigint",
-				"parent_id": "bigint",
-				"node_type": "string",
-				"created_on": "timestamp",
-				"created_by": "bigint",
-				"modified_on": "timestamp",
-				"modified_by": "bigint",
-				"version_number": "bigint",
-				"file_handle_id": "bigint",
-				"name": "string",
-				"is_public": "boolean",
-				"is_controlled": "boolean",
-				"is_restricted": "boolean"
-			},
-			"partitionKeys": {
-				"snapshot_date": "date"
+			"description" : "This table contain snapshots of nodes (Nodes are project, file, folders...). Snapshots are taken when nodes created, updated or deleted. Note: Snapshots are also taken periodically and independently of the changes. The snapshot_timestamp records when the snapshot was taken.",
```

Comment:
is there a column that indicates this: "Snapshots are also taken periodically and independently of the changes" so we can query these snapshots vs nodes that are created, updated, or deleted

Is it correct to assume:

* Everytime a node is created/updated/deleted, a row is added
* Periodically, a snapshot is created of all the nodes in the system adding N number of rows where N is the number of node that exist in Synapse.

Reply:
There is no column indicating that it's a periodic snapshot or change snapshot.

Yes, every-time a node is created/updated/deleted, a row is added
The periodic snapshots are a byproduct of migration. We replay all the messages after migrating, so a message goes out (like it did for the original operation) and a snapshot is taken (i.e. a duplicate snapshot of the last operation on each node). So periodic snapshots add one duplicate of last change(CREATE, UPDATE, DELETE) on node